src/dsql/DdlNodes.epp

```diff
@@ -12183,15 +12189,19 @@ void GrantRevokeNode::checkGrantorCanGrantDdl(thread_db* tdbb, jrd_tra* transact
 	{
 		if ( (PRV.RDB$USER_TYPE == obj_sql_role) && !attachment->att_user->roleInUse(tdbb, PRV.RDB$USER))
 			continue;
+
 		if (PRV.RDB$GRANT_OPTION == WITH_GRANT_OPTION)
+		{
 			grantable = true;
```

Comment:
I do not like an idea of leaving FOR/END_FOR loop using break. This keeps busy resources used by request.

Reply:
It was in the past (explicit isc_unwind_request call was needed), it's no longer a problem since FB3.